src/App.tsx

```diff
@@ -1,26 +1,249 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import * as todoServices from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { FilterBy } from './types/FilterBy';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { Errors } from './types/Errors';
+import { ErrorNotification } from './components/ErrorNotification';
 
-const USER_ID = 0;
+function filterTodos(todos: Todo[], show: string) {
+  switch (show) {
+    case FilterBy.Active: {
+      return todos.filter(todo => !todo.completed);
+    }
 
-export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+    case FilterBy.Completed: {
+      return todos.filter(todo => todo.completed);
+    }
+
+    case FilterBy.All:
+
+    default:
+      return todos;
   }
+}
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState(Errors.No_Error);
+  const [filterBy, setFilterBy] = useState(FilterBy.All);
+  const [isLoadedIDs, setIsLoadedIDs] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [editingTodoId, setEditingTodoId] = useState<number | null>(null);
+
+  const isFocusAddFormTrigger = useRef(false);
+
+  useEffect(() => {
+    isFocusAddFormTrigger.current = false;
+  });
+
+  useEffect(() => {
+    todoServices
+      .getTodos()
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+      })
+      .catch(() => {
+        setErrorMessage(Errors.Load);
+      });
+  }, []);
+
+  const filteredTodos = useMemo(
+    () => filterTodos(todos, filterBy),
+    [todos, filterBy],
+  );
+
+  const handleDeleteTodo = useCallback((todoId: number) => {
+    return todoServices
+      .deleteTodo(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+      })
+      .catch(error => {
+        setErrorMessage(Errors.Delete);
+        throw new Error(error);
```

Comment:
```suggestion
```
this line kill FE , we shouldn't go this way , you already handle errors with state